src/common/gitservice/blame.go

```
@@ -19,16 +21,50 @@ type GitBlame struct {
 	GitUserEmail  string
 }
 
-func NewGitBlame(filePath string, lines structure.Lines, blameResult *git.BlameResult, gitOrg string, gitRepository string, userEmail string) *GitBlame {
-	gitBlame := GitBlame{GitOrg: gitOrg, GitRepository: gitRepository, BlamesByLine: map[int]*git.Line{}, FilePath: filePath, GitUserEmail: userEmail}
+func GetPreviousBlameResult(g *GitService, filePath string) (*git.BlameResult, *object.Commit) {
+	if g.repository == nil {
+		return nil, nil
+	}
+	ref, err := g.repository.Head()
+	if err != nil {
+		return nil, nil
+	}
+	commit, err := g.repository.CommitObject(ref.Hash())
+	if err != nil {
+		return nil, nil
+	}
+	parentIter := commit.Parents()
+	previousCommit, err := parentIter.Next()
+	if err != nil {
+		return nil, nil
+	}
+
+	var previousBlameResult *git.BlameResult
+	result, _ := g.PreviousBlameByFile.Load(filePath)
```

ignoring the error is not recommended